gateway/src/main/java/org/georchestra/gateway/accounts/admin/AbstractAccountsManager.java

```diff
@@ -76,6 +80,16 @@ public GeorchestraUser getOrCreate(@NonNull GeorchestraUser mappedUser) throws D
         return find(mappedUser).orElseGet(() -> createIfMissing(mappedUser));
     }
 
+    public Optional<GeorchestraUser> findByEmailAndOAuth2OrgId(GeorchestraUser mappedUser) {
+        Optional<GeorchestraUser> user = null;
+        // search user by email
+        if ((null != mappedUser.getOAuth2Provider()) && (null != mappedUser.getOAuth2Uid())
+                && (null != mappedUser.getEmail())) {
+            user = findByEmail(mappedUser.getEmail());
+        }
+        return user;
```

Comment:
Why using Optional, if you set it to null ? I'd have written the method this way:

```
public Optional<GeorchestraUser> findByEmailAndOAuth2OrgId(GeorchestraUser mappedUser) {
        // search user by email
        if ((null != mappedUser.getOAuth2Provider()) && (null != mappedUser.getOAuth2Uid())
                && (null != mappedUser.getEmail())) {
            return findByEmail(mappedUser.getEmail());
        }
        return Optional.empty();
}
```

but maybe I missed something

Reply:
I will check that. Maybe a minor mistake.